mapper.go

```diff
@@ -0,0 +1,308 @@
+package tupleconv
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/tarantool/go-tarantool/datetime"
+	"github.com/tarantool/go-tarantool/decimal"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Mapper is a mapper from a certain type to any.
+type Mapper[T any] interface {
+	Map(src T) (any, error)
+}
+
+// Interface validations.
+var (
+	_ Mapper[string] = (*BoolParser)(nil)
+	_ Mapper[string] = (*UIntParser)(nil)
+	_ Mapper[string] = (*IntParser)(nil)
+	_ Mapper[string] = (*FloatParser)(nil)
+	_ Mapper[string] = (*DecimalParser)(nil)
+	_ Mapper[string] = (*UUIDParser)(nil)
+	_ Mapper[string] = (*StringParser)(nil)
+	_ Mapper[string] = (*DateTimeParser)(nil)
+	_ Mapper[string] = (*MapParser)(nil)
+	_ Mapper[string] = (*SliceParser)(nil)
+	_ Mapper[string] = (*BinaryParser)(nil)
+	_ Mapper[string] = (*NullParser)(nil)
+)
+
+// FuncMapper is a function-based Mapper.
+type FuncMapper[T any] struct {
+	mapFunc func(T) (any, error)
+}
+
+// NewFuncMapper creates FuncMapper.
+func NewFuncMapper[T any](mapFunc func(T) (any, error)) *FuncMapper[T] {
+	return &FuncMapper[T]{mapFunc: mapFunc}
+}
+
+// Map is the implementation of Mapper for FuncMapper.
+func (mp *FuncMapper[T]) Map(src T) (any, error) {
+	return mp.mapFunc(src)
+}
+
+// MakeSequenceMapper makes a sequential Mapper from a Mappers list.
+func MakeSequenceMapper[T any](mappers []Mapper[T]) Mapper[T] {
+	return NewFuncMapper(func(src T) (any, error) {
+		for _, mp := range mappers {
+			if result, err := mp.Map(src); err == nil {
+				return result, nil
+			}
+		}
+		return nil, NewErrUnexpectedValue(src)
+	})
+}
+
+// replaceSeparators replaces ignore characters and decimal separators.
+func replaceSeparators(src, ignoreChars, decSeparators string) string {
+	for _, char := range ignoreChars {
+		src = strings.ReplaceAll(src, string(char), "")
+	}
+	for _, char := range decSeparators {
+		src = strings.ReplaceAll(src, string(char), ".")
+	}
+	return src
+}
+
+// BoolParser is a parser to bool.
+type BoolParser struct{}
+
+// NewBoolParser creates BoolParser.
+func NewBoolParser() *BoolParser {
+	return &BoolParser{}
+}
+
+// Map is the implementation of Mapper[string] for BoolParser.
+func (*BoolParser) Map(src string) (any, error) {
+	return strconv.ParseBool(src)
+}
```

Comment:
Considering that mappers should not change an own internal state, it would be cleaner not to use pointers here.
```suggestion
// BoolParser is a parser to bool.
type BoolParser struct{}

// MakeBoolParser creates BoolParser.
func MakeBoolParser() BoolParser {
	return BoolParser{}
}

// Map is the implementation of Mapper[string] for BoolParser.
func (BoolParser) Map(src string) (any, error) {
	return strconv.ParseBool(src)
}
```

Reply:
Changed.